test/register.test.ts

```diff
@@ -0,0 +1,88 @@
+import { JSDOM } from "jsdom"
+import { loggerModule } from "./helpers"
+
+describe("register", () => {
+  describe("when navigator and serviceWorker are defined", () => {
+    const mockRegisterFn = jest.fn()
+
+    beforeAll(() => {
+      const { window } = new JSDOM()
+      global.window = (window as unknown) as Window & typeof globalThis
+      global.document = window.document
+      global.navigator = window.navigator
+      global.location = window.location
+
+      Object.defineProperty(global.navigator, "serviceWorker", {
+        value: {
+          register: mockRegisterFn,
+        },
+      })
+    })
+
+    beforeEach(() => {
+      jest.mock("@coder/logger", () => loggerModule)
+    })
```

Comment:
I hope it's okay that this test mocks here in the `beforeEach`.  When I move it outside the `describe` blocks and use the `require` method like you suggested for `constants` etc. @code-asher - it doesn't work :( 

I couldn't figure out why, but this way works!

Reply:
Yup this is perfect. In this case since we `require` within each test and the mocks are cleared after each test we also have to mock it before each test.

For the other test the require (import) happens at the top of the file, so we had to mock at the top of the file as well.

Reply:
Oh! I understand now. Thanks for explaining that! Let me know if there's anything else you need from me to move this forward!